### PR TITLE
[core] Setup e2e tests

### PR DESCRIPTION
**Pull request by DanailH**
This PR aims to provide the ability to test real browser interaction based on this pr https://github.com/mui-org/material-ui/pull/25405

I've added a simple test case to illustrate that the setup is working.

### CODE REVIEW

**Comment by dtassone** [comment]
is it running in the CI?

**Review by dtassone — APPROVED** [review]

File: test/e2e/fixtures/KeyboardNavigation/GridFocus.tsx [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    },
+    {
+      id: 2,
+      brand: 'Puma',
+    },
+  ],
+  columns: [{ field: 'brand', width: 100 }],
+};
+
+export default function BaseGridFocus() {
```
Comment (oliviertassinari) [inline test/e2e/fixtures/KeyboardNavigation/GridFocus.tsx:22]: I don't think that the file structure is flying. Material-UI X while host more components. I think that a prefix with the name of the component and a flat structure after would work a lot better. So 👍 for ` test/e2e/fixtures/DataGrid/KeyboardNavigationFocus.tsx`

File: test/e2e/index.test.ts [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    }
+  });
+
+  after(async () => {
+    await browser.close();
+  });
+
+  describe('<DataGrid />', () => {
+    it('should select cell 0-0 when pressing tab key', async () => {
+      await renderFixture('KeyboardNavigation/GridFocus');
```
Comment (oliviertassinari) [inline test/e2e/index.test.ts:66]:
```suggestion
      await renderFixture('DataGrid/KeyboardNavigationFocus');
```

**Review by oliviertassinari — CHANGES_REQUESTED** [review]
> is it running in the CI?

No

File: test/e2e/template.html [hunk 3]
```diff
@@ -0,0 +1,16 @@
+<!DOCTYPE html>
+<html>
+  <head>
+    <title>vrtest</title>
```
Comment (oliviertassinari) [inline test/e2e/template.html:4]:
```suggestion
    <title>Playwright end-to-end test</title>
```

File: test/e2e/fixtures/KeyboardNavigation/GridFocus.tsx (hunk 1 above)
Comment (DanailH) [reply test/e2e/fixtures/KeyboardNavigation/GridFocus.tsx:22]: Yes, I was not convinced about the naming. I'll update it

Comment (DanailH) [reply test/e2e/fixtures/KeyboardNavigation/GridFocus.tsx]: Updated

**Comment by DanailH** [comment]
> > is it running in the CI?
> 
> No

Should I add a new section in the `.circleci/config.yml` to make it work?

File: test/e2e/template.html (hunk 3 above)
Comment (oliviertassinari) [reply test/e2e/template.html]: I will apply this in the core repo in my next batch of small changes

**Comment by oliviertassinari** [comment]
> Should I add a new section in the .circleci/config.yml to make it work?

Yes, we need a new CI step

**Review by oliviertassinari — APPROVED** [review]
👍

Side notes for the future:

- We might be able to share some of the visual regression and end-to-end set up with the main repo. So improvements that @eps1lon bring in the core repo can benefit the whole codebase. But maybe it's too early
- The test assertion seems to be a weak spot. `expect(await page.evaluate(()` adds quite some boilerplate.
- We could have Storybook render all the content inside `test/e2e/fixtures/`. These new tests are already rendered by the end-to-end infra but maybe having them inside Storybook could serve use cases as well. Not sure, it could also slow the storybook even more.

**Pull request merged**